crates/uv/src/commands/project/install_target.rs

```diff
@@ -259,27 +259,22 @@ impl<'lock> InstallTarget<'lock> {
             Self::Project { lock, .. }
             | Self::Workspace { lock, .. }
             | Self::NonProjectWorkspace { lock, .. } => {
+                // `provides-extra` was added in Revision 1.
+                if lock.revision() == 0 {
+                    return Ok(());
+                }
+
```

Comment:
This needs to check for the current lock.version() too, since 3.0 > 2.1 (unless you want 2.100 => 3.101 to be a thing..? that seems weird.)

Reply:
Added in https://github.com/astral-sh/uv/pull/11500/commits/6571d2ffc4f1a320d875ea1f2e5994ad9b2fb414

Reply:
I think this should be `if (lock.version(), lock.revision()) < (1, 1)` rather than hard-coding 1?

Reply:
I think it's actually impossible to get here since we reject lockfiles that don't match the major, but you're right that we should change this.

Reply:
Agree on both points